flocker/ca/_ca.py

```diff
@@ -0,0 +1,225 @@
+                "keyid:always", issuer=cert
+            )
+        ])
+        cert.sign(self.keypair.original, digest)
+        return Certificate(cert)
+
+
+def generate_keypair():
+    """
+    Create a new key pair.
+    """
+    return FlockerKeyPair(
+        keypair=KeyPair.generate(crypto.TYPE_RSA, size=4096)
+    )
+
+
+class CertificateAuthority(PRecord):
+    """
+    A certificate authority whose configuration is stored in a specified
+    directory.
+
+    :ivar FilePath path: A ``FilePath`` representing the absolute path of
+        a directory containing the CRT and KEY files.
+    :ivar Certificate certificate: A signed certificate, populated only by
+        loading from path.
+    :ivar FlockerKeyPair keypair: A private/public keypair, populated only by
+        loading from path.
+    """
+    certificate = field(mandatory=True, initial=None)
+    keypair = field(mandatory=True, initial=None)
+    path = field(mandatory=True)
+
+    @classmethod
+    def from_path(cls, path):
+        """
+        :param FilePath path: Directory where private key and certificate are
+            stored.
+        """
+        if not path.isdir():
+            raise PathError(
+                b"Path {path} is not a directory.".format(path=path.path)
+            )
+
+        certPath = path.child("cluster.crt")
+        keyPath = path.child("cluster.key")
+
+        if not certPath.isfile():
+            raise PathError(
+                b"Certificate file {path} does not exist.".format(
+                    path=certPath.path)
+            )
+
+        if not keyPath.isfile():
+            raise PathError(
+                b"Private key file {path} does not exist.".format(
+                    path=certPath.path)
+            )
+
+        try:
+            certFile = certPath.open()
+        except IOError:
+            raise PathError(
+                (b"Certificate file {path} could not be opened. "
+                 b"Check file permissions.").format(
+                    path=certPath.path)
+            )
+
+        try:
+            keyFile = keyPath.open()
+        except IOError:
+            raise PathError(
+                (b"Private key file {path} could not be opened. "
+                 b"Check file permissions.").format(
+                    path=keyPath.path)
+            )
+
+        certificate = Certificate.load(
+            certFile.read(), format=crypto.FILETYPE_PEM)
+        keypair = FlockerKeyPair(
+            keypair=KeyPair.load(keyFile.read(), format=crypto.FILETYPE_PEM)
+        )
+
+        return cls(path=path, certificate=certificate, keypair=keypair)
+
+    @classmethod
+    def initialize(cls, path, name):
+        """
+        Generate new private/public key pair and self-sign, then store in
+        given directory.
+
+        :param FilePath path: Directory where private key and certificate are
+            stored.
+        :param bytes name: The name of the cluster.
```

Comment:
What would the cluster name be, typically? Maybe a little more explanation needed.